src/common/permission/PermissionManager.cpp

```diff
@@ -0,0 +1,223 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "permission/PermissionManager.h"
+
+namespace nebula {
+namespace permission {
+
+// static
+bool PermissionManager::canReadSpace(session::Session *session, GraphSpaceID spaceId) {
+    if (!FLAGS_enable_authorize) {
+        return true;
+    }
+    if (session->isGod()) {
+        return true;
+    }
+    bool havePermission = false;
+    switch (session->roleWithSpace(spaceId)) {
+        case session::Role::GOD :
```

Comment:
What is the difference between  `session->isGod()`   and session::Role::GOD ？

Reply:
> What is the difference between `session->isGod()` and session::Role::GOD ？

`session->isGod` just check if the user is god. 
`session->roleWithSpace(spaceId) `check what role of in this space.
`case session::Role::GOD` still there because the compilation requirement of switch block .